src/vaadin-radio-button.html

```diff
@@ -251,7 +255,13 @@
          * @protected
          */
         click() {
-          this.shadowRoot.querySelector('input').click();
+          // If you change this block, please test manually that radio-button and
+          // radio-group still works ok on iOS 12/13 and up as it may cause
+          // an issue that is not possible to test programmatically.
+          // See: https://github.com/vaadin/vaadin-radio-button/issues/140
+          if (!this.disabled) {
+            this.shadowRoot.querySelector('input').dispatchEvent(new MouseEvent('click'));
```

Comment:
1. Please consider using `focusElement` here.

2. Should the synthetic event have `bubbles: true` etc, same as normal event does?

Reply:
1. I thought about it but imo it's better to not use it here since `focusElement` is only defined for `ControlStateMixin` and it's about focus behaviour which is unrelated to this code so I'd like to keep it separate even though it happens to refer to the same element that we need here atm (to not so tightly couple code of unrelated behaviours).

We could have some other generic helper for getting the native input but I wanted to keep my changes minimal and related to fixing this issue,

2. I don't think it matters here since all existing tests pass and it seems to work as intended (by manual testing). I noticed currently even the number of click events triggered by checking the radio button varies from 1-3 events depending on platform (even before my changes) and some timing issues. It would make sense to try to keep it as 1 click in all cases but I'm not sure if it a problem or not. If we want to fix that there should be another issue about it and we would need tests for that also. Again I wanted to keep the code for fixing this critical bug as minimal and this seemed to work so I saw no need to add complexity to this code for now.